### ISSUE TITLE
Add a progress bar for the downloads

### ISSUE THREAD

**Issue by Flowrey** [Flowrey/youtube-bz#9]
### Problem to solve

Currently, there is no indication of the progression of a download

### Proposal

Add a progress bar to see the progression

**Comment by yr1404** [comment]
I would like to give it a try. Can you assign this issue to me?

**Comment by Flowrey** [comment]
Yeah, sure !

**Comment by yr1404** [comment]
**Code Details**

I have used tqdm for adding progress bar
```
from tqdm import tqdm
```

Added a pbar global variable
```
global pbar
    pbar = tqdm(total=len(results))
    
    # Run download in thread pool to avoid blocking IO
    for result in results:
        if result:
            loop = asyncio.get_running_loop()
            loop.run_in_executor(
                None, download_video_audio, result["title"], result["id"], destination
            )

    pbar.close()
```

And, updating the pbar here:
```
def download_video_audio(title: str, video_id: str, destination: Optional[str] = None):
    """Download audio of a YouTube video."""
    print("[Downloading] {} : {}".format(title, video_id))
    if stream := pytube.YouTube(
        f"http://youtube.com/watch?v={video_id}"
    ).streams.get_audio_only():
        stream.download(output_path=destination)
    print("[Downloaded] {}".format(title))
    pbar.update(1)
```

I'm encountering an issue where multiple progress bars are being displayed. 
Can you give me a heads up at where I might be approaching wrong?

**Comment by Flowrey** [comment]
Hey, i think the `global` here is what make you have multiple progress bars, we don't need one.
The idea is to update the progress bar when a `Future` completed.

You  would need to use the asyncio module of tqdm, `tqdm.asyncio`.
And with that you should be able to do something like this:

```python
from tqdm.asyncio import tqdm
```
```python
   loop = asyncio.get_running_loop()

    # Run download in thread pool to avoid blocking IO
    tasks = []
    for result in results:
        if result:
            tasks.append(loop.run_in_executor(
                None, download_video_audio, result["title"], result["id"], destination
            ))

    for f in tqdm.as_completed(tasks):
        await f
```

**Comment by Flowrey** [comment]
Fixed by #26
